src/utils/cachingAssets.ts

```diff
@@ -65,21 +68,28 @@ export const writeUrlInCache = ({ url, fileType, parentId, client }: WriteUrlInC
   })
 }
 
-export const writeVideoDataInCache = ({ data, thumbnailUrl, client }: WriteVideoDataCacheArg) => {
+export const writeVideoDataInCache = ({ edge, thumbnailUrl, client }: WriteVideoDataCacheArg) => {
   const video = client.cache.writeFragment({
-    id: `Video:${data.id}`,
+    id: `Video:${edge.node.id}`,
     fragment: VideoFieldsFragmentDoc,
     fragmentName: 'VideoFields',
     data: {
-      ...data,
+      ...edge.node,
       thumbnailPhotoUrls: thumbnailUrl ? [thumbnailUrl] : [],
       thumbnailPhotoAvailability: AssetAvailability.Accepted,
     },
   })
   client.cache.modify({
     fields: {
-      videos: (existingVideos = []) => {
-        return [video, ...existingVideos]
+      videosConnection: (existingVideos = {}) => {
+        return {
+          ...existingVideos,
+          pageInfo: {
+            ...existingVideos.pageInfo,
+          },
+          totalCount: existingVideos.totalCount + 1,
+          edges: [{ ...edge, node: video }, ...(existingVideos?.edges ? existingVideos.edges : [])],
+        }
       },
     },
   })
```

Comment:
Below here, we also need to make change to `removeVideoFromCache` to modify `videosConnection` query as well